batch.rst

```diff
@@ -0,0 +1,156 @@
+.. _batch:
+
+==========
+Batch Jobs
+==========
+
+Flux supports batch jobs. Users can submit batch scripts
+into Flux, which are queued up and then run when they can be
+allocated to the requested resources. Flux, however, strays away
+from the traditional approach in that batch jobs are
+supported not only at the multi-user system level
+but also at the nested single-user levels.
+
+.. note::
+   In the following, batch jobs will be described using single-user Flux,
+   but it is important to note that the same commands and techniques
+   will also work at the system level.
+
+--------------
+Launching Flux
+--------------
+
+For this demonstration, we first launch Flux under SLURM and get an interactive shell.
+
```

Comment:
Add note:

> Note: If launching under Slurm is not possible or convenient, a single-node single user instance of Flux can be started with `flux start -s 4`.

Reply:
I wonder adding this to the note may mislead some folks though. Is this something we want to advertise to end users or is this mostly for testing?

Reply:
I think it is very useful for playing around with Flux interfaces without "wasting" batch resources or if you do not have easy access to Slurm or other batch launcher. I leave it up to you if you want to include this or not.

However, I will say I believe this single node method to start Flux for testing or experimentation is one of the great features that sets Flux's usability above other RMs. I think normal users could end up using it more than we might initially think.

Reply:
Perhaps we should add a note on the availability of this stand alone mode as well as clear differences between srun `flux start` vs. flux start. 

I remember some users try `flux start`with no starter process within a batch allocation and got confused. And I also remember they added `flux start -s <N>` with the start processes and got confused.

A fuller note should be useful after all. 

Thanks!

Reply:
Yeah, even if it is just for testing, I think it is pretty useful to be able to execute ``flux start flux mini run -n4 hostname`` and similar commands from a login node. I think there are plenty of users who aren't very good at getting interactive batch sessions (plenty don't know about ``salloc`` at all), so it's especially important. Most of the important concepts you can get just by fooling around on a login node without involving anything heavyweight. I suppose there is some danger that someone would try to run an HPC application on the login node, though...

And yeah, I think one of the confusing things for users will be the difference between ``flux start CMD`` and ``flux start``, because the second one makes it _look_ like nothing happened---like flux is suddenly just running in the same shell as before---and if you just copy your terminal session into a batch script, it won't work as expected.

Reply:
Thanks @jamescorbett for chiming in!